### ISSUE TITLE
Stored credentials for a site but doesn't appear to be actually stored, LOGIN menu acts strange & MP reset

### ISSUE THREAD

**Issue by 2CNK** [limpkin/mooltipass#155]
MP Extension: 0.2.0
MP App: 0.13.0
Firmware: I can't remember where to find this, but pretty sure it's the one before RC3.2 because I haven't applied RC3.2 yet.  Sorry I can't be clearer.

Problem: 
Stored credentials on MP for site, but conflicting behaviour concerning whether credentials exist or not.  Cannot use LOGIN on MP for the site in question (to have MP send credentials manually).

While repeating the problem my MP reset back to the splash screen and locked.

Setup:
I recently created a new account (for Blizzard's Battle.net) and entered my credentials at the web page.    https://us.battle.net/login/en/. I was interrupted before hitting LOGIN on the web page.  I returned some time later and clicked LOGIN (the fields were already filled with my email and password which I typed earlier.)

The web site said the session had timed out, but the Chrome extension still asked if I wanted to save the credentials.  I did.  The MP appeared to act normally (e.g. asking for confirmation) at this point but I didn't take special notice of anything other than at least one confirmation dialog happened (there should have been more than one for a new site but I can't be sure if there was or wasn't)

App Log:  
<pre>> requesting to write credentials for battle.net
> adding new service: battle.net</pre>

I tried to log in again by refreshing the login page but the MP doesn't seem to recognize having the credentials.  App Log:
<pre>> requesting credentials for battle.net
access denied or no credentials</pre>

Trying to use LOGIN on the MP does the following strange behaviour, which is repeatable:
1. Touch LOGIN on MP
2. Scroll to 'battle.net' (2nd page, top left), touch to select
3. MP returns to home screen
4. Touch LOGIN on MP again 
5. Scroll to 'battle.net' (2nd page, top left), touch to select
6. MP asks to send credential for 'canadapost.ca' (3rd page, top left) instead

If a short delay (5 seconds?) is between steps 3 and 4 then the 'wrong' credential will not get sent, the MP just goes back to home (skips # 6).  If selecting the 'battle.net' credential again is done within a few moments, the 'wrong' credential (canadapost.ca) is done.

After repeating this process a few times investigating the timing, my MP reset to the splash screen and was locked.

**Comment by limpkin** [comment]
Hey there!

thanks for the detailed info!
the firmware version should be displayed in the app, under the log tab when you first launch it.
Are you using linux or windows? Did you approve website adding only or website + login + password?

Mathieu

**Comment by 2CNK** [comment]
Thanks for the reminder on firmware version!  

Firmware: RC3 (I was expecting a decimaled number and didn't recognize this as firmware version, duh)
OS: Windows 7 64 bit

When I was approving adding the credential I took no special notice of which confirmations were there at the time (I wasn't in 'troubleshooting' mode yet as nothing had gone wrong.)

The website adding had a confirmation, there may not have been any others (login, password) but I am not sure.

I'll try removing the site using memory management mode and try reproducing from scratch.

**Comment by 2CNK** [comment]
Ah, no go.  Memory Management mode in the app doesn't display a full list of credentials - and battle.net isn't one of the ones listed.

**Comment by limpkin** [comment]
Perfect !
If you haven't added any credential for this website the going back to menu
behaviour is normal.
Perhaps I'll add some information message...

**Comment by limpkin** [comment]
Is your mooltipass correctly behaving except that?
Do you have access to a Linux machine?

**Comment by 2CNK** [comment]
The behaviour makes sense if only the site but no actual credentials exist.

The unexpected parts were:
- Selecting the site again from the LOGIN menu actually triggered the wrong credential instead the second time around if you tried again.
- Repeating this process reset and locked the MP

Sadly I have no Linux machine handy but I could fire up a Raspberry Pi if there is something specific to try.

**Comment by 2CNK** [comment]
Well, in an attempt to investigate a bit more I changed my MP's state.  I tried a login again with an expired login page just like before, but I did not delete the site credential first.  I thought maybe I could observe the site and/or credentials saving part better.  But the app and MP worked normally this time (possibly because the site was already in the MP?) and it stored login and password A-OK.

I could try a new smartcard (I have a spare) instead if I need to re-do this (i.e. use the expired login form to see if everything repeats) if there is value in trying that.

**Comment by limpkin** [comment]
A new smartcard won't change anything.
Can you see the full list of your credentials in the app now?
Are you still getting the unexpect behaviour?

**Comment by 2CNK** [comment]
I had thought a new smartcard with a new pin, etc as though I were a new user might allow me to re-do the process from scratch.

I am getting normal behaviour now and also see all the credentials in memory management mode.

I deleted the site and credentials and will see if I can re-create.  If not, then all is well and I'll close the issue.  :D

**Comment by limpkin** [comment]
Oh trying to replicate the bug with a new user would be actually quite awesome!

**Comment by 2CNK** [comment]
I used a new smartcard and new user, I can confirm that the MP only confirmed adding the site (it did not ask confirmation to add a login or password) when I used a sessoin expired login page for battle.net.

I was also able to replicate the issue originally posted, but not the MP reboot.  That is to say, after adding a few more credentials into my MP the following will occur when selecting the site with no credentials from the LOGIN function:

- Touch LOGIN on MP
- Select "battle.net" (top left quadrant on my MP)

One of two things will happen:

- Exit back to home screen (normal)
- MP asks "Send login for ebay,ca?" (a different site)

However I think this issue can be closed for the following reasons:

1. I have a considerably harder time triggering the "wrong" site when using LOGIN function - so much that I feel this is probably a stray capacitance issue where the touch pad simply gets triggered incorrectly.  Possibly related to multiple re-triggerings.  The original issue consistently had credentials for a site on an entirely different "page" showing up instead, but I can't replicate that now.

2. I have been unable to cause the MP hardware to crash back to the splash screen and lock as a result of selecting the 'empty' site with no credentials when using LOGIN.  This happened on the original issue report but I have not been able to replicate it since.

If you're in agreement then I think this can be closed :)

**Comment by limpkin** [comment]
Hey There,

Can you update to RC4? It should provide us with more debug info... and
hopefully resolve the bug once and for all :)

Cheers,
Mathieu

**Comment by 2CNK** [comment]
Updated to RC4 and tried to replicate issue.  

Adding the site-only credential happens as usual when using the session expired login form on the web page, but it's handled perfectly well in attempts to use the MP's LOGIN now.  I have normal behaviour.  All is well there!

One thing I noticed:
Memory Management doesn't list the 'empty' site only credential in the list though so as a workaround to get rid of it it needs to be populated first with a login, then it can be removed using memory management.

I think Memory Management not showing a site-only entry without login credentials attached to it was why I thought management mode wasn't showing everything.
